command.go

```diff
@@ -216,6 +216,12 @@ func (c *Command) Context() context.Context {
 	return c.ctx
 }
 
+// SetContext replaces the underlying command context so that parent
+// commands can pass down values to their subcommands.
+func (c *Command) SetContext(ctx context.Context) {
```

Comment:
How is this any different from `ExecuteContext`? With using `ExecuteContext`, a user could pass in the necessary parent context, and then it is executed on the child sub-command.

Reply:
If I'm not mistaken (it's been a while since this PR was submitted), by using `ExecuteContext` you're not able to set the context in the scope of a command so that nested commands have access to the new, modified context. In other words, the context is only passed to the root command, and then it is simply forwarded without modification to any subcommands. If you want parent commands to be able to act as middleware to their children, you need to have a structure that can be modified within the root context, which might not be desirable. That's where `SetContext` comes in handy. It allows to modify the context at will.

Reply:
@jpmcb Any update? :) Thanks!